concent_api/core/message_handlers.py

```diff
@@ -1312,27 +1322,51 @@ def handle_send_subtask_results_verify(
             reason=message.concents.ServiceRefused.REASON.InvalidRequest,
         )
 
-    (requestor_has_enough_deposit, provider_has_enough_deposit) = bankster.claim_deposit(
+    subtask = get_one_or_none(
+        Subtask,
+        subtask_id=compute_task_def['subtask_id'],
+    )
+
+    if subtask is not None and subtask.state_enum in [
+        Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER,
+        Subtask.SubtaskState.ADDITIONAL_VERIFICATION,
+    ]:
+        return message.concents.ServiceRefused(
+            reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+        )
+
+    (claim_against_requestor, claim_against_provider) = bankster.claim_deposit(
+        subtask_id=task_to_compute.subtask_id,
         concent_use_case=ConcentUseCase.ADDITIONAL_VERIFICATION,
         requestor_ethereum_address=task_to_compute.requestor_ethereum_address,
         provider_ethereum_address=task_to_compute.provider_ethereum_address,
         subtask_cost=task_to_compute.price,
+        requestor_public_key=requestor_public_key,
+        provider_public_key=provider_public_key,
     )
 
-    if not requestor_has_enough_deposit:
+    if claim_against_requestor is None:
         return message.concents.ServiceRefused(
             reason=message.concents.ServiceRefused.REASON.TooSmallRequestorDeposit,
         )
-    if not provider_has_enough_deposit:
+    if claim_against_provider is None:
         return message.concents.ServiceRefused(
             reason=message.concents.ServiceRefused.REASON.TooSmallProviderDeposit,
         )
 
+    if subtask is not None and subtask.state_enum in [
+        Subtask.SubtaskState.ACCEPTED,
+        Subtask.SubtaskState.FAILED,
+    ]:
```

Comment:
Here you're first creating claims and then immediately deleting them if we're in a state where verification is not allowed. Why not make this check before creating claims instead?

Reply:
:+1: done